ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -71,41 +77,53 @@
   private final ObjectMapper mapper;
   private final StatementParser statementParser;
   private final KsqlEngine ksqlEngine;
-  private final ServiceContext serviceContext;
   private final CommandQueue commandQueue;
   private final ListeningScheduledExecutorService exec;
   private final ActivenessRegistrar activenessRegistrar;
   private final QueryPublisher queryPublisher;
   private final PrintTopicPublisher topicPublisher;
   private final Duration commandQueueCatchupTimeout;
   private final TopicAccessValidator topicAccessValidator;
+  private final KsqlSecurityExtension securityExtension;
+  private final ServiceContextFactory serviceContextFactory;
 
   private WebSocketSubscriber<?> subscriber;
 
+  @VisibleForTesting
+  @FunctionalInterface
+  interface ServiceContextFactory {
```

Comment:
This doesn't feel like the right place to put this but I wasn't sure where it belonged. Suggestions appreciated!